python/shotgun_desktop/upgrade_startup.py

```diff
@@ -48,6 +48,9 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     if not _supports_get_from_location_and_paths(sgtk):
         return False
 
+    if 'SGTK_DESKTOP_DISABLE_UPDATE' in os.environ:
```

Comment:
We're in process of revamping the Shotgun Desktop startup code (just look at the `develop/zero_config` branch, the changes are massive and there's more to come) and part of that means looking into better ways of blocking updates to the desktop startup.

In the meantime, instead of introducing a new environment variable, maybe you could use the `SGTK_DESKTOP_STARTUP_LOCATION` environment variable? It points to a location on disk and forces the Shotgun Desktop to use that copy of the desktop startup code and [disables auto-updates](https://github.com/bpabel/tk-framework-desktopstartup/blob/429bebbb796b9c23aa74c28e4b72598a05b302fe/python/shotgun_desktop/upgrade_startup.py#L65-L67), which removes the need to connect to the Toolkit app store. You can then put a copy of that package on your network and have every user of the Shotgun Desktop point to that using the environment variable. If you ever need a bug fix for that package, you can simply pull it down from Github and copy it at the same location on the network.